### PR TITLE
refactor out zig bug workarounds that are fixed

### PR DESCRIPTION
**Pull request by nektro**

### CODE REVIEW

File: src/logger.zig [hunk 1]
```diff
@@ -562,8 +560,6 @@ pub const Msg = struct {
     }
 };
 
-// Do not mark these as packed
```
Comment (Jarred-Sumner) [inline src/logger.zig]: Please leave this comment in. This costed me a day.

File: src/js_ast.zig [hunk 2]
```diff
@@ -7371,9 +7371,7 @@ pub const Macro = struct {
                 }
             };
 
-            // TODO: can change back to `return CallData.callWrapper(.{`
-            // when https://github.com/ziglang/zig/issues/16242 is fixed
-            return CallData.callWrapper(CallArgs{
+            return CallData.callWrapper(.{
```
Comment (Jarred-Sumner) [inline src/js_ast.zig:7374]: Please manually verify this works as expected. @dylan-conway probably remembers this one well

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
These changes look innocuous but I don’t trust our test coverage adequately checks this doesn’t cause regressions

If you want to merge this PR, please manually test each workaround has been fixed

File: src/js_ast.zig (hunk 2 above)
Comment (dylan-conway) [reply src/js_ast.zig:7374]: To verify this one, you will need to verify the args in `runAsync` are the same as the input passed through `callWrapper`. If I remember correctly, some of the fields in the `CallArgs` struct would become garbage. Probably best way to test is run macros with 0-n args

Comment (dylan-conway) [reply src/js_ast.zig:7374]: I should've added a test with this commit

**Comment by jdalton** [comment]
In my old job we'd call this "trust work". I'm very thankful @nektro for this!

**Pull request closed without merging**